enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,487 @@
+  instances in a cluster placement group.
+
+##### Spread Placement Groups
+
+A Spread placement group is a grouping of instances all placed in distinct racks within the datacenter.
+This ensures that each instance within the group has its own network and power source, maximising resillience to
+hardware failures.
+
+Spread placement groups are recommended for launching a small number of instances that must be kept separate, eg a
+cluster of etcd instances.
+
+As Spread placement groups keep instances in different racks, they can be expanded over time and can also mix instance
+types without detriment.
+
+Spread placement groups have a limited capacity. You can only launch seven instances within a Spread placement group
+within a single availability zone.
+
+###### Limitations of Spread Placement Groups
+
+- Spread placement groups are not supported for AWS Dedicated Instances.
+
+##### Partition Placement Groups
+
+A Partition placement group divides your instances into logical segments called partitions.
+Each partition is assigned a set of racks within the Availability Zone, each rack in turn having its own network and
+power source.
+No rack within the AZ will appear in more than one partition.
+
+Users may either create instances and have them spread automatically across partitions or they may allocate instances
+to a particular partition.
+Partitions are identified numerically, indexed from 1.
+
+Partition placement groups are recommended for deploying large distributed and replicated workloads across distinct
+racks.
+
+Partition placement groups are limited to seven partitions per Availability Zone.
+
+When using a Partition placement group, if users wish to control the placement of their workloads within Partitions,
+they will need to create multiple MachineSets and assign each one to a partition.
+Once they have assigned different MachineSets to partitions, they will be able to use Node affinity to assign the
+workloads to the Nodes created by the Machines within the MachineSet by selecting based on the MachineSets unique
+labels.
+
+###### Limitations of Partition Placement Groups
+
+- A partition placement group with AWS Dedicated Instances can have a maximum of two partitions.
+
+#### Creation of Placement Groups
+
+To enable users to get started with Placement Groups, we will allow users to specify the name of the placement group to
+use for the new Machines.
+If the placement group does not exist, we will create it based on the configuration provided by the user.
+
+We will allow the user to specify the type of the placement group to create, either Cluster, Spread or Partition, as
+well as, in the case of Partition, the number of partitions to be created.
+
+Should the user not define the type, we will default to Spread to ensure maximal hardware fault tolerance within the
+group.
+
+Should the user not define the number of partitions, we will omit this value in the request and leave AWS to default
```

Comment:
why do we do this instead of forcing them to consider it?  it's an advanced feature and to use it, they have to have a matching number of machinesets.  if we don't require a value, I don't know how we expect users to know aws's default.

Reply:
> it's an advanced feature and to use it, they have to have a matching number of machinesets.

They don't actually *have* to create a matching number of MachineSets. It is supposedly also valid to create a partition placement group and have AWS automatically spread instances between the partitions.

I would expect a user using this to have thought about it and decided on a partition count having read the AWS docs, but perhaps they don't care? Both our docs will call out the AWS default and the AWS docs already do